korlibs-concurrent/src/korlibs/concurrent/lock/Lock.kt

```diff
@@ -0,0 +1,64 @@
+@file:Suppress("PackageDirectoryMismatch")
+
+package korlibs.concurrent.lock
+
+import korlibs.concurrent.thread.*
```

Comment:
* imports are not adviced since they make it hard to determine which classes are from which imports.

Reply:
Rationale behind the star imports:

KorGE uses star imports because it relies a lot on extension members. It was a pain in the ass adding all those members manually for people starting. Users typically don't care where classes come from as long as the code works.

Reply:
Yes, but it makes it pain in the ass to refactor. For our current task forexample it is important to see which methods are used from which packages, when we want to reduce the dependency towards a package.

Reply:
Why it is a pain in the ass? Doesn't `@Deprecated` has a parameter to specify replacement to do it automatically inside the IDE?